core-plugins/src/main/java/io/cdap/plugin/batch/action/EmailAction.java

```diff
@@ -107,37 +109,39 @@ public Config() {
       includeWorkflowToken = false;
     }
 
-    public void validate() {
-      super.validate();
+    public void validate(FailureCollector collector) {
+      super.validate(collector);
 
       if (!containsMacro("username") && (Strings.isNullOrEmpty(username) ^ Strings.isNullOrEmpty(password))) {
-        throw new IllegalArgumentException("You must either set both username and password or " +
-                                             "neither username nor password.");
+        collector.addFailure("You must either set both username and password or neither username nor password.",
```

Comment:
this was an existing problem, but most places we don't use 'You'. Change this to:

Both username and password must be given, or neither of them must be given.

Reply:
Done